src/Microsoft.AspNetCore.JsonPatch/Internal/DictionaryAdapterOfTU.cs

```diff
@@ -126,6 +127,47 @@ public class DictionaryAdapter<TKey, TValue> : IAdapter
             return true;
         }
 
+        public bool TryTest(
+            object target,
+            string segment,
+            IContractResolver contractResolver,
+            object value,
+            out string errorMessage)
+        {
+            var contract = (JsonDictionaryContract)contractResolver.ResolveContract(target.GetType());
+            var key = contract.DictionaryKeyResolver(segment);
+            var dictionary = (IDictionary<TKey, TValue>)target;
+
+            if (!TryConvertKey(key, out var convertedKey, out errorMessage))
+            {
+                return false;
+            }
+
+            // As per JsonPatch spec, the target location must exist for remove to be successful
+            if (!dictionary.ContainsKey(convertedKey))
+            {
+                errorMessage = Resources.FormatTargetLocationAtPathSegmentNotFound(segment);
+                return false;
+            }
+
+            if (!TryConvertValue(value, out var convertedValue, out errorMessage))
+            {
+                return false;
+            }
+
+            var currentValue = dictionary[convertedKey];
+            if (!(JsonConvert.SerializeObject(currentValue) == JsonConvert.SerializeObject(convertedValue)))
```

Comment:
We might want to use `JObject` here. Also, the contract resolver needs to be involved somehow

Reply:
The key being used has already been resolved with `contract.DictionaryKeyResolver(segment)` https://github.com/aspnet/JsonPatch/pull/114/files/8763258935a6eb2b7c151cda519884bf6f8dd041#diff-eba89d20437ba32b8dca5690b028526bR138